opensafely/codelists.py

```diff
@@ -184,6 +185,13 @@ def parse_codelist_file(codelists_dir):
                 f"{line} does not match [project]/[codelist]/[version] "
                 "or user/[username]/[codelist]/[version]"
             )
+        codelist_id = "/".join(tokens[:-1])
+        if (existing_version := codelist_versions.get(codelist_id)) is not None:
```

Comment:
question: Should we also check for the same version, i.e. redundant duplicated lines? We could here.

Reply:
I think we should be encouraging cleanliness, and having duplicated lines isn't very clean. I don't think this is critical for acceptance of this change.

Reply:
I agree. Though looking again: the current behaviour here actually ends up giving a misleading error — if you have a duplicate, you'll get an error that it's a different version, so I'll revisit this.